tango_simlib/tango_sim_generator.py

```diff
@@ -505,26 +504,33 @@ def configure_device_model(sim_data_file=None, test_device_name=None):
     for file_name in data_file:
         parsers.append(get_parser_instance(file_name))
 
+    # In case there is more than one device instance per class.
+    models = {}
+    if dev_names:
+        for dev_name in dev_names:
+            models[dev_name] = Model(dev_name)
+    else:
+        models[dev_name] = Model(dev_name)
+
     # In case there is more than one parser instance for each file
-    model = Model(dev_name)
-    command_info = {}
-    properties_info = {}
-    override_info = {}
-    for parser in parsers:
-        PopulateModelQuantities(parser, dev_name, model)
-        command_info.update(parser.get_device_command_metadata())
-        properties_info.update(parser.get_device_properties_metadata('deviceProperties'))
-        override_info.update(parser.get_device_cmd_override_metadata())
-    PopulateModelActions(command_info, override_info, dev_name, model)
-    PopulateModelProperties(properties_info, dev_name, model)
-    
-    return model
+    for model in models.values():
+        command_info = {}
+        properties_info = {}
+        override_info = {}
+        for parser in parsers:
+            PopulateModelQuantities(parser, dev_name, model)
```

Comment:
`dev_name` needs to be extracted from the `models` dict - it is not being updated by the outer loop.